skbio/stats/power.py

```diff
@@ -1073,7 +1073,13 @@ def _draw_paired_samples(meta_pairs, index, num_samps):
     subs = []
 
     # Draws the other groups
-    for set_, num_ in collections.Counter(set_pos).items():
+    # for set_, num_ in collections.Counter(set_pos).items():
+    counter = collections.Counter(set_pos)
+
+    # collections.Counter(set_pos).items() order isn't guaranteed
```

Comment:
No. The problem arises when compiling examples. In example, a `np.seed(25)` is instantiated, so you can reproduce always your sample code and `np.random.choice` will be the same. The example reported in #1497 is derived from your examples: the point is that `collections.Counter(set_pos).items()` change order in different python version, `_draw_paired_samples` returns a different array and `_identify_sample_groups` fails example tests